chainer/testing/matrix.py

```diff
@@ -44,6 +44,10 @@ def generate_matrix(shape, dtype=float, **kwargs):
     if (singular_values < 0).any():
         raise ValueError('negative singular value is given')
 
+    if 0 in shape:
```

Comment:
Could you check the shape of `singular_values` (or leave it as `# TODO`)?

Reply:
Ok, let me take some time to think about how to do that.

Reply:
Fixed and added a test case to ensure the check.